src/components/DialogContent.tsx

```diff
@@ -17,16 +17,36 @@ limitations under the License.
 import * as React from "react";
 import { ModuleApi } from "../ModuleApi";
 import { PlainSubstitution } from "../types/translations";
+import { ModuleUiDialogOptions } from "../types/ModuleUiDialogOptions";
 
+/** Props for {@link DialogContent} */
 export interface DialogProps {
+    /**
+     * A reference to the active Module API .
+     */
     moduleApi: ModuleApi;
+
+    /**
+     * Updates the options of the dialog.
+     * @param options - The updates that should be applied to the dialog options.
+     */
+    setOptions(options: Partial<ModuleUiDialogOptions>): void;
```

Comment:
I'm struggling to understand how this is populated, or how it is used.

Is it a callback that the module has to provide when it calls `openDialog`?

Reply:
ok, so where is the implementation of `setOptions` in this example?

Reply:
That is a task for the matrix-react-sdk: https://github.com/matrix-org/matrix-react-sdk/pull/11395/files#diff-39867f2e1adc7a387871eedac805d542f29418aee2162f4d1c6ae991bec39a5dR71-R80